From a79d648fcde72fc98048d4435bc86864a59fd01b Mon Sep 17 00:00:00 2001
From: loqs <loqs@users.noreply.github.com>
Date: Mon, 20 Nov 2017 23:46:52 +0000
Subject: [PATCH] Allow the elevator parameter to apply to an MQ queue.

---
 block/elevator.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/block/elevator.c b/block/elevator.c
index c197271694af2..c6175b0eabe8e 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -225,12 +225,12 @@ int elevator_init(struct request_queue *q, char *name)
 	}
 
 	/*
-	 * Use the default elevator specified by config boot param for
-	 * non-mq devices, or by config option. Don't try to load modules
+	 * Use the default elevator specified by config boot param,
+	 * or by config option for non-mq devices. Don't try to load modules
 	 * as we could be running off async and request_module() isn't
 	 * allowed from async.
 	 */
-	if (!e && !q->mq_ops && *chosen_elevator) {
+	if (!e && *chosen_elevator) {
 		e = elevator_get(q, chosen_elevator, false);
 		if (!e)
 			printk(KERN_ERR "I/O scheduler %s not found\n",
