---
 kernel/sched/bfs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-3.16-ck1/kernel/sched/bfs.c
===================================================================
--- linux-3.16-ck1.orig/kernel/sched/bfs.c	2014-08-16 14:46:11.960916841 +1000
+++ linux-3.16-ck1/kernel/sched/bfs.c	2014-08-19 08:49:17.237171535 +1000
@@ -5195,7 +5195,7 @@ out:
 	task_grq_unlock(&flags);
 
 	if (running_wrong)
-		_cond_resched();
+		__cond_resched();
 
 	return ret;
 }
